src/Analyzers/Core/Analyzers/OrderModifiers/AbstractOrderModifiersDiagnosticAnalyzer.cs

```diff
@@ -25,12 +34,24 @@ internal abstract class AbstractOrderModifiersDiagnosticAnalyzer : AbstractBuilt
             : base(IDEDiagnosticIds.OrderModifiersDiagnosticId,
                    option,
                    language,
-                   new LocalizableResourceString(nameof(FeaturesResources.Order_modifiers), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
-                   new LocalizableResourceString(nameof(FeaturesResources.Modifiers_are_not_ordered), FeaturesResources.ResourceManager, typeof(FeaturesResources)))
+                   new LocalizableResourceString(nameof(AnalyzersResources.Order_modifiers), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)),
+                   new LocalizableResourceString(nameof(AnalyzersResources.Modifiers_are_not_ordered), AnalyzersResources.ResourceManager, typeof(AnalyzersResources)))
         {
             _syntaxFacts = syntaxFacts;
             _option = option;
             _helpers = helpers;
+            _diagnosticPropertiesByPreferredOrder = new ConcurrentDictionary<string, ImmutableDictionary<string, string>>();
+        }
+
+        private ImmutableDictionary<string, string> GetDiagnosticProperties(string preferredOrderString)
```

Comment:
We embed the PreferredOrder string in the diagnostic properties now so the fixer does not need to fetch option value from DocumentOptionSet. We also cache the dictionaries by the string values to reduce allocations.

Reply:
FYI: this change was not needed and I was able to revert in by just switching the CodeFixer to use a different API to fetch option, which is available in both CodeStyle and Workspaces layer. I have forced pushed the changes to reflect this.